### PR TITLE
[ON HOLD] Remove unsafe-inline and unsafe-eval from Content Security Policy

### PR DESCRIPTION
**Pull request by chmreid**
This modifies the Content Security Policy to remove unsafe-eval and unsafe-inline from the script-src directive. This will likely lead to more script violations of the CSP. Follow-up work will expand the script-src policy to allow scripts as needed.

Also removing unsafe-hashes directive, as it requires the allowed hashes to be specified in the CSP header itself.

### CODE REVIEW

**Review by david4096 — APPROVED** [review]

**Review by coverbeck — APPROVED** [review]

**Comment by chmreid** [comment]
Before merging this in, I want to raise a potential issue that could cause a flood of CSP style violations.

When I inspect the contents of the dev.dockstore.net homepage, I see a lot of greyed-out `<style>` tags (exactly the kind of element that will be blocked by adding this directive). I don't know what the greyed-out text style means in Firefox, and am wondering if those elements will trigger multiple CSP violations.

If we're not sure whether those elements are going to cause CSP violations, we can deploy this, and check the new CSP reports to see if each element causes CSP violations. If so, we can roll back this PR and figure out how to deal with that.

Screen Shot 2021-04-01 at 11 01 32

**Review by garyluu — APPROVED** [review]

**Review by denis-yuen — APPROVED** [review]

**Review by GFJHogue — APPROVED** [review]

**Pull request closed without merging**
